peer/connector.go

```diff
@@ -151,8 +197,30 @@ fail:
 	return errX
 }
 
+// combine multi error into one
+func compositeError(errors []error) error {
+	if nil == errors || 0 == len(errors) {
+		return nil
+	}
+	var ce strings.Builder
+	ce.WriteString("composite error: [")
```

Comment:
What the output will be like ? I think [ ] is for the model.  What is the output of this? At first look compositeError is to combine several errors together. Where we have several errors what kind of format should we show to user?  what is the error should show ? (not display)

Reply:
The format is like `composite error: [error1, error2, error3]`
The previous implementation uses single routine so that it will throws error when the first connection is failed. When i separate into multi goroutines, i will collect all error from the connections and return to the caller. That why i need to declare a composite error that combine multi error into one.

Reply:
I agree with you in case whether we need to stop program if one of the connections is failed or continue working and ignore it, as long as we have at least 1 static connection. I think the second one is better. But when all of them is failed, we need to print out the error of all failed connection, not only one. So the `CompositeError` is necessary to show it's combine of a lot of errors. Of course we can use `GenericError` but it's more common and i don't expect it.